src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -210,15 +224,22 @@
   (async/go (try
               (loop []
                 (let [task (async/<! lifecycle-chan)]
-                  (if (= task :shutdown)
-                    (do
-                      (log/debug "Received shutdown command, lifecycle worker exiting.")
-                      :done)
-                    (do
-                      (log/debug "Lifecycle worker executing lifecycle task.")
-                      (task)
-                      (log/debug "Lifecycle worker completed lifecycle task; awaiting next task.")
-                      (recur)))))
+                  (schema/validate LifeCycleTask task)
+                  (let [{:keys [type task-function]} task]
+                    (condp #(contains? %1 %2) type
+                      #{:shutdown}
+                      (do
+                        (log/debug "Received shutdown command, lifecycle worker exiting.")
+                        :done)
+
+                      #{:boot :restart}
+                      (do
+                        (log/debugf "Lifecycle worker executing %s lifecycle task." type)
+                        (task-function)
```

Comment:
Good catch.  In this PR, the 'shutdown' path doesn't have a task-function, which is why I had to do the schema that way.  In a later PR when I move the shutdown work into the go loop, the shutdown case gains a task-function.  So I think what would make sense to me is to leave this as-is in this PR but then just help make sure that I harden the schema in the follow-up PR?